### PR TITLE
ll40ls cleanup and create PX4Rangerfinder helper class

### PR DESCRIPTION
**Pull request by dagar**
@mcsauder would you be able to give this a quick review and test (if you have any version of the lidar lite)?

Saves more than 2 kilobytes of flash on fmu-v2.

### CODE REVIEW

File: src/lib/drivers/CMakeLists.txt [hunk 1]
```diff
@@ -39,5 +39,6 @@ add_subdirectory(gyroscope)
 add_subdirectory(led)
 add_subdirectory(linux_gpio)
 add_subdirectory(magnetometer)
+add_subdirectory(rangefinder)
```
Comment (mcsauder) [inline src/lib/drivers/CMakeLists.txt]: @dagar , this PR is looking good.  Would you be opposed to renaming this directory distance_sensor to match the `src/drivers/distance_sensor` directory name/structure, (and the class name as well)?

Comment (mcsauder) [reply src/lib/drivers/CMakeLists.txt]: One more thought, the `ToneAlarmInterface` might be a good naming/design pattern to follow.  perhaps a `distance_sensor` directory with a `DistanceSensorInterface` class would be appropriate.

**Comment by mcsauder** [comment]
@dagar ,

Data on the bench looks like PX4:master, basic functionality all looks good:
image


I have a few little thoughts, like keeping the naming convention adhered to some kind of standard, but I think getting this PR in and following with minor adjustments later is the right approach.

This PR looks good to me. Depending on how you feel about the directory naming convention, I'd suggest merging this PR as-is and iterating any remaining improvements from there.

Nice work!

File: src/drivers/distance_sensor/ll40ls/CMakeLists.txt [hunk 2]
```diff
@@ -40,4 +40,6 @@ px4_add_module(
 		LidarLite.cpp
 		LidarLiteI2C.cpp
 		LidarLitePWM.cpp
+	DEPENDS
+		drivers_rangefinder
 	)
```
Comment (mcsauder) [inline src/drivers/distance_sensor/ll40ls/CMakeLists.txt]: Hi @dagar, can you add `STACK_MAIN 1200` to the CMakeLists.txt? (Or allocate more if you prefer.)

**Comment by mcsauder** [comment]
Hi @dagar , it would be good to get PR #12453 in before this as well.

File: src/lib/drivers/CMakeLists.txt (hunk 1 above)
Comment (dagar) [reply src/lib/drivers/CMakeLists.txt]: Having sensor in the name started to feel inconsistent with the others (accel, gyro, differential_pressure).

**Comment by dagar** [comment]
Rebased on master (with https://github.com/PX4/Firmware/pull/12453 merged), stack main set to 1500, and copyright year updated.

**Comment by dagar** [comment]
Saves 2.1 kB of flash on fmu-v2.

image

**Pull request merged**